src/HTTP/UrlClient.cpp

```diff
@@ -54,12 +54,11 @@ class cUrlClientRequest:
 		m_Callbacks->OnError(a_ErrorMessage);
 
 		// Terminate the request's TCP link:
-		auto link = m_Link;
+		auto link = m_Link.lock();
 		if (link != nullptr)
```

Comment:
Just learned  about `std:.weak_ptr`.
Now I've also learned the nice trick `if(auto link = m_Link.lock())` from SO 😄.
Maybe you also could use this to limit the scope of the `link` variable.